server/datastore/mysql/mdm.go

```diff
@@ -1588,11 +1590,13 @@ func (ds *Datastore) AreHostsConnectedToFleetMDM(ctx context.Context, hosts []*f
 }
 
 func (ds *Datastore) IsHostConnectedToFleetMDM(ctx context.Context, host *fleet.Host) (bool, error) {
-	mp, err := ds.AreHostsConnectedToFleetMDM(ctx, []*fleet.Host{host})
-	if err != nil {
-		return false, ctxerr.Wrap(ctx, err, "finding if host is connected to Fleet MDM")
+	if host.Platform == "windows" {
+		return isWindowsHostConnectedToFleetMDM(ctx, ds.reader(ctx), host)
+	} else if host.Platform == "darwin" || host.Platform == "ipados" || host.Platform == "ios" {
+		return isAppleHostConnectedToFleetMDM(ctx, ds.reader(ctx), host)
 	}
-	return mp[host.UUID], nil
+
+	return false, nil
```

Comment:
LGTM, the single-host logic could've been done in `AreHostsConnectedToFleetMDM` (with a check for len(hosts) == 1) to catch perhaps a few more calls, but the hot paths ones are probably calling the single-host method directly anyway.

Reply:
Yep. I evaluated that as an option but decided to go this way because it was a little less risky, given that the multi-host path indexes by UUID and I needed to index by integer PK to be safe for non-parameterized queries. From what I could see, the hot paths for a single host all called the single-host function.